### PR TITLE
Adds missing autopilot snapshot test and avoids snapshotting nil.

### PR DESCRIPTION
**Pull request by slackpad**
Just some small things I noticed while working nearby.

### CODE REVIEW

File: agent/consul/state/autopilot_test.go [hunk 1]
```diff
@@ -92,3 +93,47 @@ func TestStateStore_AutopilotCAS(t *testing.T) {
...
+
+		idx, res, err := s.AutopilotConfig()
+		if err != nil {
+			t.Fatalf("err: %s", err)
+		}
+		if idx != 99 {
+			t.Fatalf("bad index: %d", idx)
+		}
+		verify.Values(t, "", before, res)
+	}()
```
Comment (kyhavlov) [inline agent/consul/state/autopilot_test.go]: Why put all this in an anonymous function?

Comment (slackpad) [inline agent/consul/state/autopilot_test.go]: I pasted it - lemme get rid of that since it is unnecessary.
